### PR TITLE
alpine: explicitly install libretls and zlib

### PR DESCRIPTION
**Pull request by gdams**
fixes: https://github.com/adoptium/containers/issues/136

paging @DRoppelt to verify that this is the required code change

### CODE REVIEW

File: dockerfile_functions.sh [hunk 1]
```diff
@@ -225,7 +225,7 @@ print_alpine_pkg() {
 # Select the alpine OS musl based packages.
 print_alpine_musl_pkg() {
 	cat >> "$1" <<'EOI'
-RUN apk add --no-cache tzdata musl-locales musl-locales-lang \
+RUN apk add --no-cache libretls musl-locales musl-locales-lang tzdata zlib \
```
Comment (karianna) [inline dockerfile_functions.sh:228]: Worth commenting the why for each of these packages.

**Review by karianna — APPROVED** [review]

**Comment by DRoppelt** [comment]
Oh this totally fell under the table that I wanted to submit a PR, sorry for that.

Yes seems about right. 

out of curiosity, since this is where I left off when I tried to understand where changes would need to go:

I would have expected to see changes in the jdk/jre Dockerfiles (https://github.com/adoptium/containers/blob/main/17/jdk/alpine/Dockerfile.releases.full#L24), where/how will that be updated? Shouldnt changes be checked in with this PR or is that some sort of post-merge that automatically updates these files?

The CI pipelines seem to run with the old install versions: https://github.com/adoptium/containers/runs/6753868952?check_suite_focus=true

```

Step 1/7 : FROM alpine:3.15
 ---> 0ac33e5f5afa
Step 2/7 : ENV LANG='en_US.UTF-8' LANGUAGE='en_US:en' LC_ALL='en_US.UTF-8'
 ---> Running in 8ab0e8a5ed41
Removing intermediate container 8ab0e8a5ed41
 ---> a8063ba055ea
Step 3/7 : RUN apk add --no-cache tzdata musl-locales musl-locales-lang     && rm -rf /var/cache/apk/*
 ---> Running in e0745f931a38
fetch https://dl-cdn.alpinelinux.org/alpine/v3.15/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.15/community/x86_64/APKINDEX.tar.gz
(1/3) Installing musl-locales (0_git20201005-r0)
(2/3) Installing musl-locales-lang (0_git20201005-r0)
(3/3) Installing tzdata (2022a-r0)
Executing busybox-1.34.1-r5.trigger
```

While I do not expect these changes to cause issues, the CI/CD pipelines did not actually verify that these changes were correct, is that right?

**Comment by gdams** [comment]
@DRoppelt the dockerfiles are updated automatically once this PR is merged. I'll merge this now and a PR will appear within the next 10 minutes that will be automatically merged once the CI completes.

**Comment by zmyjs** [comment]
```
Sending build context to Docker daemon  17.28MB
Step 1/8 : FROM nodered/node-red:latest
 ---> e7631c7bedde
Step 2/8 : USER root
 ---> Using cache
 ---> 2ebb802ddb14
Step 3/8 : ENV LANG='en_US.UTF-8' LANGUAGE='en_US:en' LC_ALL='en_US.UTF-8'
 ---> Using cache
 ---> a8980f763438
Step 4/8 : RUN apk add --no-cache libretls musl-locales musl-locales-lang tzdata zlib     && rm -rf /var/cache/apk/*
 ---> Running in cad983ce67ee
fetch http://dl-cdn.alpinelinux.org/alpine/v3.12/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.12/community/x86_64/APKINDEX.tar.gz
ERROR: unable to select packages:
  libretls (no such package):
    required by: world[libretls]
The command '/bin/sh -c apk add --no-cache libretls musl-locales musl-locales-lang tzdata zlib     && rm -rf /var/cache/apk/*' returned a non-zero code: 1
```
apk source does not have the package libretls, resulting in an error

**Comment by zmyjs** [comment]
@gdams 

Maybe I used Alpine V3.12, cause this problem.

**Pull request merged**